app/models/quote.rb

```diff
@@ -18,7 +20,13 @@ class Quote < ActiveRecord::Base
   private
 
   def reporter_and_user_differ
-    return unless reporter_id == user_id
+    return unless reporter_id == user_id && !anonymous
+
+    errors.add(:user, ": Je kan toch niet jezelf quoten!")
+  end
+
+  def anonymous_user_and_reporter_nil
+    return unless anonymous && (user_id || reporter_id)
```

Comment:
Met present? is het meteen duidelijk dat je checkt of ze gezet zijn 
```suggestion
    return unless anonymous? && (user.present? || reporter.present?)
```